c++/src/kj/async-inl.h

```diff
@@ -2063,8 +2063,39 @@ concept NoWaitScope = !isSameType<Decay<T>, WaitScope>();
 // Define a Concept to use in our `coroutine_traits` specialization to validate allowable coroutine
 // parameter types.
 
+struct DisallowedWithinCoroutine {};
+
+template <typename T, typename = int>
+struct isDisallowedInCoroutine {
```

Comment:
Nitpick: Types should start with a capital letter.

Typically I will declare a sibling function like:

```
template <typename T>
inline constexpr bool isDisallowedInCoroutine<T>() {
  return IsDisallowedInCoroutine<T>::value;
}
```

This is just for syntax sugar, so callers don't have to add `::value`.